cubi_tk/sodar_api.py

```diff
@@ -0,0 +1,151 @@
+import argparse
+from collections import namedtuple
+import os
+from typing import Literal
+
+from logzero import logger
+import requests
+
+from .common import is_uuid, load_toml_config
+from .exceptions import ParameterException, SodarAPIException
+
+
+# FIXME: maybe this should be used as a MixIn for other functions?
+class SodarAPI:
+    def __init__(self, sodar_url: str, sodar_api_token: str, project_uuid: str):
+        self.sodar_url = sodar_url
+        self.sodar_api_token = sodar_api_token
+        self.project_uuid = project_uuid
+        self.check_args()
+
+    def check_args(self):
+        # toml_config needs an object with attribute named config
+        args = namedtuple("args", ["config"])
+        toml_config = load_toml_config(args(config=None))
+        if not self.sodar_url:
+            if not toml_config:
+                raise ParameterException(
+                    "SODAR URL not given on command line and not found in toml config files."
+                )
+            self.sodar_url = toml_config.get("global", {}).get("sodar_server_url")
+            if not self.sodar_url:
+                raise ParameterException(
+                    "SODAR URL not found in config files. Please specify on command line."
+                )
+        if not self.sodar_api_token:
+            if not toml_config:
+                raise ParameterException(
+                    "SODAR API token not given on command line and not found in toml config files."
+                )
+            self.sodar_api_token = toml_config.get("global", {}).get("sodar_api_token")
+            if not self.sodar_api_token:
+                raise ParameterException(
+                    "SODAR API token not found in config files. Please specify on command line."
+                )
+        if not is_uuid(self.project_uuid):
+            logger.warning("Project UUID does not look like valid UUID.")
```

Comment:
Would it make sense to error here instead of warn? If the project uuid is not a uuid, it will probably error downstream anyway, right?

Reply:
We can do that yes. testing will have to use an dummy UUID, but that's not a bad thing, I guess